wordpress-mobile/WordPress-Android
# MSD - Stats Card - Create UI with mocked data

Parent #15201 
References https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2244

The PR adds a `Today's Stats` card in the `My Site Dashboard` using the data from the `mocked_site_dashboard_cards.json` stored locally. Once the `API` changes for the new endpoint are ready, `Today's Stats` data will be consumed from the `API` and the local `JSON` will be deleted. For now, the card will display static data, navigation, and touch events will be taken up later in a different PR. 

### UI 
| Light Mode | Dark Mode | 
| :---: | :---: | 
| ![Screenshot_20220201-063920](https://user-images.githubusercontent.com/17463767/151922938-b2e7c33d-1cb7-476f-88c3-377b4633a9f9.jpg) |![Screenshot_20220201-063907](https://user-images.githubusercontent.com/17463767/151922974-8dbd6c83-1c2d-44c1-91d4-2e11a21ebc2b.jpg) | 

UI cases 
| Maximum font size | Portuguese | French  | Arabic(RTL) |
| :---: | :---: | :---: | :---: | 
| ![Screenshot_20220201-064557](https://user-images.githubusercontent.com/17463767/151923155-e758bffd-442b-42ac-a25a-0d7057437c1a.jpg)|![Screenshot_1643703152](https://user-images.githubusercontent.com/17463767/151933599-8581b136-ccb4-49ea-9192-c2d984e8325f.png)|![Screenshot_20220201-084229](https://user-images.githubusercontent.com/17463767/151923358-649f7e83-df1c-4ed3-b05c-163b32b392eb.jpg)|![Screenshot_٢٠٢٢٠٢٠١-٠٨٤٣٢٥](https://user-images.githubusercontent.com/17463767/151923359-d0b47ecc-99f9-4a0f-a25d-07fc78346ce2.jpg)|


To test:
Prerequisites
1. Launch the app
2. Select a wp com site 
3. Navigate to My Site -> App Settings -> Debug Settings
4. Ensure that the `MySiteDashboardPhase2FeatureConfig` is to set to on
5. Restart the app if needed
6. Navigate to the My Site tab
7. Notice that Today's stats card is shown.

### Known Accessibility Issue 
The order in which Voice over announces the `Today's Stats` card contents are as follows
Today's stats title -> Views title-> Visitors title -> Likes title-> Views count -> Visitors count -> Likes count

The expected order is 
Today's stats title-> Views title-> Views count -> Visitors title ->  Visitors count -> Likes title -> Likes count 

This improvement is added in the parent issue as a Todo and will be taken up in a separate PR later.


## Regression Notes
1. Potential unintended areas of impact
- Displays Today's stats card if the flag is not enabled
- Displays Today's stats card for Self hosted site 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual tests

3. What automated tests I added (or what prevented me from doing so)
- Added tests in `TodaysStatsCardBuilderTest.kt`
- Added tests in `CardsSourceTest.kt`
- Added tests in `CardsBuilderTest.kt`

PR submission checklist:
